ambry-protocol/src/main/java/com.github.ambry.protocol/BlobStoreControlAdminRequest.java

```diff
@@ -35,21 +36,31 @@
    */
   public static BlobStoreControlAdminRequest readFrom(DataInputStream stream, AdminRequest adminRequest)
       throws IOException {
-    Short versionId = stream.readShort();
-    if (!versionId.equals(VERSION_V1)) {
-      throw new IllegalStateException("Unrecognized version for BlobStoreControlAdminRequest: " + versionId);
+    short versionId = stream.readShort();
+    short numReplicasCaughtUpPerPartition;
+    BlobStoreControlRequestType controlRequestType;
+    switch (versionId) {
+      case VERSION_V1:
+        numReplicasCaughtUpPerPartition = stream.readShort();
+        controlRequestType =
+            stream.readByte() == 1 ? BlobStoreControlRequestType.StartStore : BlobStoreControlRequestType.StopStore;
+        break;
+      case VERSION_V2:
```

Comment:
Is `VERSION_V2` needed? Look like `BlobStoreControlRequestType.values()[stream.readByte()]` can fully cover `VERSION_V1`.

Reply:
Thanks @zzmao @ankagrawal for your inputs here. You are right, no need to keep version here. I will simplify this piece of code.